src/utils/misc.cc

```diff
@@ -698,3 +699,75 @@ void packagekit_suggest_quit()
   ExternalProgram pkcall( argv );
   pkcall.close();
 }
+
+const char *computeStatusIndicator( const PoolItem &pi_r )
+{
+  bool isLocked = pi_r.status().isLocked();
+  ui::Selectable::Ptr sel( ui::Selectable::get( pi_r ) );
+
+  if ( pi_r->isSystem() )
+    return lockStatusTag( "i", isLocked, pi_r.identIsAutoInstalled() );
+
+  // picklist: ==> available, maybe identical installed too
+  // only check for sel->installedEmpty() if NOT pseudo installed
+  if ( !traits::isPseudoInstalled( pi_r->kind() ) && sel->installedEmpty() )
+    return lockStatusTag( "", isLocked );
+
+  bool identicalInstalledToo = ( traits::isPseudoInstalled( pi_r->kind() )
+                                   ? ( pi_r.isSatisfied() )
+                                   : ( sel->identicalInstalled( pi_r ) ) );
+  if ( identicalInstalledToo )
+    return lockStatusTag( "i", isLocked, pi_r.identIsAutoInstalled() );
+
+  return lockStatusTag( "v", isLocked );
+}
+
+const char *computeStatusIndicator(const ui::Selectable &s, const Edition &installedMustHaveEd )
+{
+  // whether to show the solvable as 'installed'
+  bool installed = false;
+
+
+  Zypper &zypper = Zypper::instance();
+  const auto &repos = zypper.runtimeData().repos;
+
+  if ( traits::isPseudoInstalled( s.kind() ) )
+    installed = s.theObj().isSatisfied();
+  // check for installed counterpart in one of specified repos (bnc #467106)
+  else if ( InitRepoSettings::instance()._repoFilter.size() )
```

Comment:
@mlandres Do we need to move that code into Selectable::identicalInstalled as well?